### PR TITLE
fix: Chart is not updating on change of variables

### PR DESCRIPTION
**Pull request by csudani7**
Close: https://github.com/SigNoz/signoz/issues/1805

### CODE REVIEW

File: frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx [hunk 1]
```diff
@@ -69,6 +74,10 @@ function VariableItem({
 		getOptions();
 	}, [getOptions]);
 
+	const onRefreshHandler = (): void => {
+		updateTimeInterval(getDefaultOption(location.pathname));
```
Comment (palashgdev) [inline frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx:78]: update time interval take selected interval and min max time in second

Comment (palashgdev) [inline frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx:78]: do not access the global if you want pathname grab from useLocation hook

**Review by palashgdev — CHANGES_REQUESTED** [review]

**Comment by palashgdev** [comment]
also check there are some eslint fix which need to be done to sort the import

**Comment by palashgdev** [comment]
Also why we are updating the global time for this one @csudani7 ?

File: frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx (hunk 1 above)
Comment (csudani7) [reply frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx]: Updated My PR With this suggested changes @palashgdev

**Comment by csudani7** [comment]
> also check there are some eslint fix which need to be done to sort the import

@palashgdev - in any specific file? I checked in my local there is no import issue i found. Can you please mention more details?

File: frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx (hunk 1 above)
Comment (csudani7) [reply frontend/src/container/NewDashboard/DashboardVariablesSelection/VariableItem.tsx]: @palashgdev 
The problem was, when we select any variables from dropdown, we needs to auto refresh manually.
So, I added refresh call on change of dropdown value. So no needs to do extrac CTA.

**Comment by csudani7** [comment]
> Also why we are updating the global time for this one @csudani7 ?

I used it for auto refresh on change of dropdown value.

**Comment by palashgdev** [comment]
@csudani7 updating the global time is not the right logic for this one..

**Comment by srikanthccv** [comment]
Since the main logic comments are not addressed yet, I will let my PR to go through review.

**Comment by csudani7** [comment]
Waiting for @srikanthccv 's PR to review: https://github.com/SigNoz/signoz/pull/2037

File: frontend/src/container/TopNav/DateTimeSelection/index.tsx [hunk 2]
```diff
@@ -254,6 +261,7 @@ function DateTimeSelection({
 		searchStartTime,
 		updateTimeInterval,
 		globalTimeLoading,
+		selectedData?.variables,
```
[the remark this answers is not in the document]
Comment (csudani7) [reply frontend/src/container/TopNav/DateTimeSelection/index.tsx:249]: Okay, Noted.

**Review by palashgdev — APPROVED** [review]

**Pull request merged**